.github/workflows/golangci-lint.yml

```diff
@@ -12,10 +12,12 @@ jobs:
     name: lint
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/setup-go@v4
+      - uses: actions/checkout@v4
+      - id: goversion
+        run: |
+          cat Dockerfile | awk '/^FROM golang:.* AS builder$/ {v=$2;split(v,a,":");split(a[2],b,"-")}; END {printf("version=%s", b[1])}' >> $GITHUB_OUTPUT
+      - uses: actions/setup-go@v5
         with:
-          go-version: '1.20'
-          cache: false
-      - uses: actions/checkout@v3
+          go-version: "${{ steps.goversion.outputs.version }}"
```

Comment:
Would anything go wrong if we [used the version from the `go.mod` file](https://github.com/marketplace/actions/setup-go-environment#getting-go-version-from-the-gomod-file)? AFAIK the one in there (as opposed to the version of the `go` binary itself) is more important to the toolchain anyway.

Reply:
I'm not sure if dependabot doesn't yet update that. Will check, though :)

Reply:
Not sure if I like this approach as it has a high potential to confuse during debugging what version is actually used. We then have two Go versions defined:

- The one in the `Dockerfile` which is mostly relevant for CI/CD and not used locally
- The `toolchain` line inside the `go.mod` which is mostly relevant for local development

If dependabot cannot update the toolchain line, then I'd be more in favor of sticking with the Dockerfile info and not putting a `toolchain` into the `go.mod` file :/ 

Still looking for information 🙂

Reply:
Well, with the initial change it would be done by dependabot as it would update the entry inside the Dockerfile. It probably didn't do so for 1.21.6 this time, though, as it is not available for alpine3.17 :) I will also bump the image to use 1.21.5-alpine so that it can be picked up by dependabot again.